src/main/scala/org/broadinstitute/hail/Utils.scala

```diff
@@ -596,17 +598,17 @@ class RichPairTraversableOnce[K, V](val t: TraversableOnce[(K, V)]) extends AnyV
   }
 }
 
-class RichPairIterator[K, V](val it: Iterator[(K, V)]) {
+class RichPairIterator[K : Ordering, V](val it: Iterator[(K, V)]) {
```

Comment:
It is too bad Scala can't unify against available implicits.  I guess it isn't tractable if there are multiple implicit parameters?

The `PK` and `project` are only needed by `localKeySort`.  It is only used by OrderedRDD.  I'd probably move it to the OrderedRDD companion object and parameterize it by the `OrderedKey` implicit.

Otherwise, I'd move it to `RichOrderedKeyPairIterator` which takes an `OrderedKey` implicit instead of a `K: Ordering` implicit.

Since `RichPairIterator` is taking a `K: Ordered` implicit, you might call it `RichOrderedIterator` or something.  (`OrderedRDDFuntions` is Spark's name for `RDD`s that are pairs where the key is ordered.)